pkg/pillar/iptables/ssh.go

```diff
@@ -95,3 +96,36 @@ func dropPortRange(startPort int, endPort int) {
 	IptableCmd("-A", "INPUT", "-p", "tcp", "--dport", portStr, "-j", "REJECT", "--reject-with", "tcp-reset")
 	Ip6tableCmd("-A", "INPUT", "-p", "tcp", "--dport", portStr, "-j", "REJECT", "--reject-with", "tcp-reset")
 }
+
+func allowControlFlows() {
+	// Allow HTTP, ssh and guacamole packets
+	// Pick flow marking values 1, 2, 3 from the reserved space.
+	portStr := "8080,22,4822"
+	IptableCmd("-t", "mangle", "-I", "PREROUTING", "1", "-p", "tcp",
+		"--match", "multiport", "--dports", portStr,
+		"-j", "CONNMARK", "--set-mark", "1")
+
+	Ip6tableCmd("-t", "mangle", "-I", "PREROUTING", "1", "-p", "tcp",
+		"--match", "multiport", "--dports", portStr,
+		"-j", "CONNMARK", "--set-mark", "1")
+
+	// Allow VNC packets
+	portStr = "5900:5999"
```

Comment:
We set up this in UpdateVncAccess() which always allows localhost vnc, and conditionally allows remote vnc. Note that this is managed by nim early in the boot - before we have an IP addresses configured. Thus I think you should put the vnc code in that function.

Reply:
We allow all these control packets from PREROUTING in raw table and let INPUT hook of filter table make the decision. Intent is to avoid dropping control packets destined to dom0 and not make filtering decisions.

Reply:
Also, the function "allocControlFlows" is called from nim.

Reply:
Same comments as below; comment is wrong based on what you say, and no rule to match to mark =1.

Also, having this mark plus having the existing reject rules which match on the ports and localhost vs not can't be the intendent endpoint, right?

Reply:
I'll re-work the comment part and function name.

Reply:
done